google-cloud-storage/src/test/java/com/google/cloud/storage/StorageImplMockitoTest.java

```diff
@@ -395,22 +419,151 @@ public void testGetOptions() {
 
   @Test
   public void testCreateBucket() {
-    when(storageRpcMock.create(BUCKET_INFO1.toPb(), EMPTY_RPC_OPTIONS))
-        .thenReturn(BUCKET_INFO1.toPb())
-        .thenThrow(new RuntimeException("Fail"));
```

Comment:
What about the exception scenario with `thenThrow(new RuntimeException("Fail"));`. I understand it was not present in existing test case but I believe we can test that scenario as well.

Reply:
Good catch! I added throwing exception on extra call in theese tests as well.

Reply:
I believe we still need to write test case around exception scenarios. I don't see i t in current test.

Reply:
Oh, I see what do you mean.
The purpose of `.doThrow(UNEXPECTED_CALL_EXCEPTION)` to check that the method is called only once.

And yes, I agree, it makes sense to add tests to verify the behavior when RPC call fails.

Thanks!

Reply:
done:
- new tests for exceptions are added
- tests for reader() improved to verify the output produced by read()